smithy-upgrade-tool/src/main/java/software/amazon/smithy/upgrade/Upgrader.java

```diff
@@ -0,0 +1,389 @@
+        ShapeUpgradeVisitor visitor = new ShapeUpgradeVisitor(completeModel, IoUtils.readUtf8File(filePath));
+
+        completeModel.shapes()
+                .filter(shape -> shape.getSourceLocation().getFilename().equals(filePath.toString()))
+                // Apply updates to the shapes at the bottom of the file first.
+                // This lets us modify the file without invalidating the existing
+                // source locations.
+                .sorted(new SourceLocationSorter().reversed())
+                .forEach(shape -> shape.accept(visitor));
+
+        return updateVersion(visitor.getModelString());
+    }
+
+    private static String updateVersion(String modelString) {
+        Matcher matcher = VERSION.matcher(modelString);
+        if (matcher.find()) {
+            return matcher.replaceFirst("\\$version: \"2.0\"\n");
+        }
+        return "$version: \"2.0\"\n\n" + modelString;
+    }
+
+    // Sorts shapes in the order they appear in the file.
+    private static class SourceLocationSorter implements Comparator<FromSourceLocation>, Serializable {
+        @Override
+        public int compare(FromSourceLocation s1, FromSourceLocation s2) {
+            SourceLocation sourceLocation = s1.getSourceLocation();
+            SourceLocation otherSourceLocation = s2.getSourceLocation();
+
+            if (!sourceLocation.getFilename().equals(otherSourceLocation.getFilename())) {
+                return sourceLocation.getFilename().compareTo(otherSourceLocation.getFilename());
+            }
+
+            int lineComparison = Integer.compare(sourceLocation.getLine(), otherSourceLocation.getLine());
+            if (lineComparison != 0) {
+                return lineComparison;
+            }
+
+            return Integer.compare(sourceLocation.getColumn(), otherSourceLocation.getColumn());
+        }
+    }
+
+    private static class TraitSorter implements Comparator<Trait>, Serializable {
+        @Override
+        public int compare(Trait o1, Trait o2) {
+            return new SourceLocationSorter().compare(o1.getSourceLocation(), o2.getSourceLocation());
+        }
+    }
+
+    private static class ShapeUpgradeVisitor extends ShapeVisitor.Default<Void> {
+        private final Model completeModel;
+        private String modelString;
+
+        ShapeUpgradeVisitor(Model completeModel, String modelString) {
+            this.completeModel = completeModel;
+            this.modelString = modelString;
+        }
+
+        String getModelString() {
+            if (!modelString.endsWith("\n")) {
+                modelString = modelString + "\n";
+            }
+            return modelString;
+        }
+
+        @Override
+        protected Void getDefault(Shape shape) {
+            if (shape.hasTrait(BoxTrait.class)) {
+                eraseTrait(shape, shape.expectTrait(BoxTrait.class));
+            }
+            // handle members in reverse definition order
+            shape.members().stream()
```

Comment:
I think this will attempt to modify list/set/map shapes, since they return their standard members via `.members()`. It's worth proving out that those are safe in the test cases.

Reply:
Yep, that's intentional since the box trait can technically show up there too. Updated to add cases for all the collection types.